src/rgw/driver/rados/rgw_rados.cc

```diff
@@ -4116,6 +4116,11 @@ int RGWRados::fetch_remote_obj(RGWObjectCtx& obj_ctx,
     bl.clear(); // overwrite source's status
     bl.append("REPLICA");
   }
+  if (zones_trace) { // add replication trace
+    bufferlist bl;
+    encode(*zones_trace, bl);
```

Comment:
at this point in `fetch_remote_obj()`, rgw_data_sync.cc has already added the local zone to the trace